### PR TITLE
Rob 31

### PR DESCRIPTION
**Pull request by SuragBalajepalli**
PR for horizontally rotating mh185 positioner xacro and moveit launch files

### CODE REVIEW

File: motoman_motopos_mh185_rotisserie_support/urdf/headstock/headstock.urdf.xacro [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        value="83.801" />
+      <inertia
+        ixx="0.089907"
+        ixy="-0.0012323"
+        ixz="-0.12263"
+        iyy="7.7723"
+        iyz="0.00031051"
+        izz="7.6922" />
+    </inertial>
+    <xacro:arg name="visible" default="true"/>
```
Comment (bkent324) [inline motoman_motopos_mh185_rotisserie_support/urdf/headstock/headstock.urdf.xacro]: I think `visible` argument is left over from Corsa as well, but we can go ahead and remove it

**Comment by SuragBalajepalli** [comment]
Made necessary changes using the package "ma1440_d500_moveit_config" as template

File: motoman_motopos_mh185_rotisserie_support/urdf/motopos_mh185_rotisserie_macro.xacro [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                           parent="world"
+                           child="${prefix}base_link"
+                           axis_xyz="0 0 0"/>
+
+
+  <xacro:fixed_joint_macro name="base-hsbase"
+                           origin_xyz="${world2pos_xyz}"
+                           origin_rpy="${world2pos_rpy}"
+                           parent="${prefix}base_link"
+                           child="hs_base_link"
```
Comment (bkent324) [inline motoman_motopos_mh185_rotisserie_support/urdf/motopos_mh185_rotisserie_macro.xacro]: We should probably prepend this `${prefix}` value to all of the links in this positioner and update the individual .xacro files as appropriate

File: motoman_ma1440_mh185_rotisserie_moveit_config/config/joint_limits.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# joint_limits.yaml allows the dynamics properties specified in the URDF to be overwritten or augmented as needed
+# Specific joint properties can be changed with the keys [max_position, min_position, max_velocity, max_acceleration]
+# Joint limits can be turned off with [has_velocity_limits, has_acceleration_limits]
+joint_limits:
+  mh185_joint_1:
+    has_velocity_limits: true
+    max_velocity: 6.28318530718
```
Comment (bkent324) [inline motoman_ma1440_mh185_rotisserie_moveit_config/config/joint_limits.yaml]: we should update this value to the max velocity listed on the datasheet

File: motoman_motopos_mh185_rotisserie_support/config/robot_calibration_offsets.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+mh185_vertical_robot_calibration_offsets:
```
Comment (bkent324) [inline motoman_motopos_mh185_rotisserie_support/config/robot_calibration_offsets.yaml]: I don't think the values from this file are actually being used anywhere

File: motoman_motopos_mh185_rotisserie_support/urdf/motopos_mh185_rotisserie_macro.xacro [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+  <!-- Add the headstock - hs_base_link, hs_revolute_link, hs_prismatic_link -->
+  <xacro:include filename="$(find motoman_motopos_mh185_rotisserie_support)/urdf/headstock/headstock.urdf.xacro" ns="hs" />
+  <xacro:include filename="$(find motoman_motopos_mh185_rotisserie_support)/urdf/tailstock/tailstock.urdf.xacro" ns="ts" />
+
+  <!-- And connect positioner_link to base_link -->
+
+  <link name="${prefix}base_link"/>
+  <link name="positioner/base"/>
```
Comment (bkent324) [inline motoman_motopos_mh185_rotisserie_support/urdf/motopos_mh185_rotisserie_macro.xacro]: we also need to add a `<link name="positioner/center"/>` frame that coincides with the `positioner/base` frame when the positioner joint angle is zero. the positioner/center frame is the frame we use to stitch the individual scans together from the structured light into the composite

File: motoman_motopos_mh185_rotisserie_support/urdf/motopos_mh185_rotisserie_macro.xacro [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+>
+
+  <xacro:property name="M_PI" value="3.1415926535897931"/>
+
+  <xacro:property name="offsets" value="${load_yaml(offsets_file_path)[prefix + 'robot_calibration_offsets']}"/>
+
+
+
+
+<xacro:property name="world_to_pos_base_tf" value="${load_yaml('$(find motoman_motopos_mh185_rotisserie_support)/config/positioner_params.yaml')}"/>
```
Comment (bkent324) [inline motoman_motopos_mh185_rotisserie_support/urdf/motopos_mh185_rotisserie_macro.xacro]: we'll eventually want to replace this file with the `robot_calibration_offsets` file we pass in from the higher level launch files, but can leave this as is for now. we'll need to do a bit of thinking on how the calibration process will be done and how to set up the frames for everything before it's worth updating

File: motoman_motopos_mh185_rotisserie_support/config/robot_calibration_offsets.yaml (hunk 4 above)
Comment (SuragBalajepalli) [reply motoman_motopos_mh185_rotisserie_support/config/robot_calibration_offsets.yaml]: Yes, the file path gets passed to the xacro but it doesn't do anything with it yet. The xacro actually has a hard coded offset it in. Could change it such that the xacro gets that hard coded offset value from this file, along with other params mentioned in 'positioner_params.yaml'

**Pull request merged**